playbooks/robusta_playbooks/alerts_integration.py

```diff
@@ -439,3 +443,76 @@ def stack_overflow_enricher(alert: PrometheusKubernetesAlert):
             )
         ]
     )
+
+
+class ForeignLogParams(LogEnricherParams):
+    """
+    :var selectors: List of specific label selectors to retrieve logs from
+    """
+
+    selectors: List[str]
+
+
+@action
+def foreign_logs_enricher(event: ExecutionBaseEvent, params: ForeignLogParams):
+    """
+    Generic enricher to fetch and attach pod logs.
+
+    The logs are fetched for the pod determined by the label selector field in the parameters.
+    """
+
+    api = client.CoreV1Api()
+    matching_pods: List[V1Pod] = []
+    for selector in params.selectors:
+        try:
+            pods: V1PodList = api.list_pod_for_all_namespaces(label_selector=selector)
+            if pods.items:
+                matching_pods = pods.items
+                break
+        except Exception as e:
+            if not (isinstance(e, exceptions.ApiException) and e.status == 404):
+                raise ActionException(
+                    ErrorCodes.ACTION_UNEXPECTED_ERROR, f"Failed to list pods for foreign log enricher.\n{e}"
+                )
+
+    if not matching_pods:
+        logging.warning(f"[foreign_logs_enricher] failed to find any matching pods for the selectors: {params.selectors}")
+
+    if matching_pods:
```

Comment:
it seems like all of this is the same code as `logs_enricher`
I would move all to its own function and use that function for here and logs enricher.
Something like this: 
`def pod_log_enricher(pod: Pod, params: LogEnricherParams) -> Optional[str]:
`
so we don't duplicate code

Reply:
Yup, i have deduplicated the code now.